components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/DataFlowProcessingEngine.cs

```diff
@@ -203,40 +240,40 @@ public class DataFlowProcessingEngine : IProcessingEngine, IDisposable {
             diagInfo.AppendLine("  # Opc endpoint connected?            : {isConnectionOk,14:0}");
             diagInfo.AppendLine("  # Monitored Opc nodes succeeded count: {goodNodes,14:0}");
             diagInfo.AppendLine("  # Monitored Opc nodes failed count   : {badNodes,14:0}");
 
             _logger.Information(diagInfo.ToString(),
-                Name,
-                TimeSpan.FromSeconds(totalDuration),
-                _messageTrigger.DataChangesCount, dataChangesPerSecFormatted,
-                _messageTrigger.ValueChangesCount, valueChangesPerSecFormatted,
-                _batchDataSetMessageBlock.OutputCount,
-                _encodingBlock.InputCount, _encodingBlock.OutputCount,
-                _messageEncoder.NotificationsProcessedCount,
-                _messageEncoder.NotificationsDroppedCount,
-                _messageEncoder.MessagesProcessedCount,
-                _messageEncoder.AvgNotificationsPerMessage,
-                _messageEncoder.AvgMessageSize, messageSizeAveragePercentFormatted,
-                chunkSizeAverage,
-                estimatedMsgChunksPerDay,
-                _batchNetworkMessageBlock.OutputCount,
-                _sinkBlock.InputCount,
-                _sinkBlockInputDroppedCount,
-                _messageSink.SentMessagesCount, sentMessagesPerSecFormatted,
-                _messageTrigger.NumberOfConnectionRetries,
-                _messageTrigger.IsConnectionOk,
-                _messageTrigger.NumberOfGoodNodes,
-                _messageTrigger.NumberOfBadNodes);
+                info.Id,
+                info.IngestionDuration,
+                info.IngressDataChanges, dataChangesPerSecFormatted,
+                info.IngressValueChanges, valueChangesPerSecFormatted,
+                info.IngressBatchBlockBufferSize,
+                info.EncodingBlockInputSize, info.EncodingBlockOutputSize,
+                info.EncoderNotificationsProcessed,
+                info.EncoderNotificationsDropped,
+                info.EncoderIoTMessagesProcessed,
+                info.EncoderAvgNotificationsMessage,
+                info.EncoderAvgIoTMessageBodySize, messageSizeAveragePercentFormatted,
+                info.EncoderAvgIoTChunkUsage,
+                info.EstimatedIoTChunksPerDay,
+                info.OutgressBatchBlockBufferSize,
+                info.OutgressInputBufferCount,
+                info.OutgressInputBufferDropped,
+                info.OutgressIoTMessageCount, sentMessagesPerSecFormatted,
+                info.ConnectionRetries,
+                info.OpcEndpointConnected,
+                info.MonitoredOpcNodesSucceededCount,
+                info.MonitoredOpcNodesFailedCount);
 
             string deviceId = _identity.DeviceId ?? "";
             string moduleId = _identity.ModuleId ?? "";
             kDataChangesCount.WithLabels(deviceId, moduleId, Name)
-                .Set(_messageTrigger.DataChangesCount);
+                .Set(info.IngressDataChanges);
```

Comment:
`info.IngressValueChanges` can be used instead of `_messageTrigger.ValueChangesCount` on line `282`.